crates/runtime/src/lib.rs

```diff
@@ -87,7 +87,7 @@ pub enum Error {
 pub type Result<T, E = Error> = std::result::Result<T, E>;
 
 pub struct Runtime {
-    pub app: Arc<RwLock<App>>,
+    pub app: Option<Arc<RwLock<App>>>,
```

Comment:
This should be `Arc<RwLock<Option<App>>>` - putting the Option first defeats the point of having the Arc & RwLock wrappers - since only one object can reference the variable with the Option.

Putting the Option inside means that we have the flexibility to replace the app inside the Arc<RwLock<>> and have the be reflected in all places that have that reference - otherwise what you end up with is several places in code with their view of App, but when its updated in one place, the other places in code that reference the App will still be referencing the old version.

Reply:
Thanks for the suggestions! I updated to Arc<RwLock<Option<App>>> and the new code is ready for review